### PR TITLE
FLW-77 , Added Option to type Name Fileds in Hindi/Assamese

### PR DESCRIPTION
**Pull request by ritik-bizbrolly**
## 📋 Description

FLW-77: 

Add Option to type Name Fileds in Hindi/Assamese

---

## ✅ Type of Change

- [ ] 🐞 **ENHANCEMENT** 

---

## ℹ️ Additional Information

For all the Name fields user can fill the data in selected language Hindi/Assamese and when selected language is in English There was a validation for checking all letter are in Caps or not.

## Summary by CodeRabbit

- **New Features**
	- Updated weight input handling to accept integer values (1 to 7000 grams) instead of decimals.
	- Simplified validation logic for weight input, enhancing user experience.

- **Bug Fixes**
	- Improved data integrity by ensuring validation processes are appropriately applied based on user language preference.
	- Streamlined navigation processes by removing unnecessary parameters and dependencies.

### CODE REVIEW

File: app/src/main/java/org/piramalswasthya/sakhi/configuration/InfantRegistrationDataset.kt [hunk 1]
```diff
@@ -111,12 +111,13 @@ class InfantRegistrationDataset(
     private var weight = FormElement(
         id = 11,
         inputType = InputType.EDIT_TEXT,
-        title = "Weight at Birth(kg)",
+        title = "Weight at Birth(gram)",
```
Comment (drtechie) [inline app/src/main/java/org/piramalswasthya/sakhi/configuration/InfantRegistrationDataset.kt:114]: What happens to old data when we make this change?

File: app/src/main/java/org/piramalswasthya/sakhi/helpers/Konstants.kt [hunk 2]
```diff
@@ -26,9 +26,9 @@ object Konstants {
     const val minAgeForNcd: Int = 30
     const val minAgeForReproductiveAge: Int = 15
     const val maxAgeForReproductiveAge: Int = 49
-    const val maxAgeForInfant: Int = 1
-    const val minAgeForChild: Int = 2
-    const val maxAgeForChild: Int = 5
+    const val maxAgeForInfant: Int = 42
```
Comment (drtechie) [inline app/src/main/java/org/piramalswasthya/sakhi/helpers/Konstants.kt:29]: These validations are off.

File: app/src/main/java/org/piramalswasthya/sakhi/configuration/InfantRegistrationDataset.kt (hunk 1 above)
Comment (sushant-bizbrolly) [reply app/src/main/java/org/piramalswasthya/sakhi/configuration/InfantRegistrationDataset.kt:114]: Yes @drtechie, for that we have to make migrations.
cc: @madhavaramu83

File: app/src/main/java/org/piramalswasthya/sakhi/helpers/Konstants.kt (hunk 2 above)
Comment (sushant-bizbrolly) [reply app/src/main/java/org/piramalswasthya/sakhi/helpers/Konstants.kt:29]: Yes, in another PR it has changed.
Checked with @ritik-bizbrolly, validations were changed twice.
cc: @madhavaramu83

**Review by NikhilFegade — APPROVED** [review]

File: app/src/main/java/org/piramalswasthya/sakhi/configuration/InfantRegistrationDataset.kt (hunk 1 above)
Comment (sushant-bizbrolly) [reply app/src/main/java/org/piramalswasthya/sakhi/configuration/InfantRegistrationDataset.kt]: @drtechie , I have reversed it back to kg, in DB we are saving it in kg, discussed with @madhavaramu83

**Pull request merged**